File: 154-find-minimum-in-rotated-sorted-array-ii/find-minimum-in-rotated-sorted-array-ii.java
class Solution {
    public int findMin(int[] nums) {
        int n = nums.length;
        int l = 0;
        int r = n-1;

        while(l<r){
            int m = (l+r)/2;
            
            

            

           if(nums[r]<nums[m]){
                l= m+1;


            }
            else if(nums[m]<nums[r]){
                r = m;

            }
            else if(nums[m]==nums[r]) r--;
            
            
        }
        return nums[l];


        
    }
}

File: 154-find-minimum-in-rotated-sorted-array-ii/find-minimum-in-rotated-sorted-array-ii.js
const findMin = nums => {
    const dnc = (l, r) => {
        if (l === r)
            return nums[l];

        if (nums[l] < nums[r])
            return nums[l];

        const m = (l + r) >> 1;

        return Math.min(
            dnc(l, m),
            dnc(m + 1, r)
        );
    };

    return dnc(0, nums.length - 1);
};